src/Microsoft.ML.Data/Model/Onnx/OnnxContext.cs

```diff
@@ -157,43 +168,103 @@ public OnnxNode CreateNode(string opType, string input, string output, string na
+        /// Call this function to declare a global double
+        /// </summary>
+        /// <param name="value">The double number which is going to be added into the ONNX graph</param>
+        /// <param name="name">A string used as a seed to generate this initializer's name in the ONNX graph.</param>
+        /// <param name="makeUniqueName">Whether a unique name should be picked for this initializer.</param>
+        /// <returns>The initializer's ONNX name</returns>
+        public abstract string AddInitializer(double value, string name = null, bool makeUniqueName = true);
+
+        /// <summary>
+        /// Call this function to declare a global ulong or uint
+        /// </summary>
+        /// <param name="value">The long number which is going to be added into the ONNX graph</param>
+        /// <param name="isUint64">true if value contains a ulong value and false if it contains uint </param>
+        /// <param name="name">A string used as a seed to generate this initializer's name in the ONNX graph.</param>
+        /// <param name="makeUniqueName">Whether a unique name should be picked for this initializer.</param>
+        /// <returns>The initializer's ONNX name</returns>
+        public abstract string AddInitializer(ulong value, bool isUint64, string name = null, bool makeUniqueName = true);
+
+        /// <summary>
+        /// Call this function to declare a global bool tensor
+        /// </summary>
+        /// <param name="values">The boolean values which are going to be added into the ONNX graph</param>
+        /// <param name="dims">The shape of values</param>
+        /// <param name="name">A string used as a seed to generate this initializer's name in the ONNX graph.</param>
+        /// <param name="makeUniqueName">Whether a unique name should be picked for this initializer.</param>
+        /// <returns>The initializer's ONNX name</returns>
+        public abstract string AddInitializer(IEnumerable<bool> values, IEnumerable<long> dims, string name = null, bool makeUniqueName = true);
+
+        /// <summary>
+        /// Call this function to declare a global float tensor
         /// </summary>
         /// <param name="values">The floats which are going to be added into the ONNX graph</param>
-        /// <param name="dims">The shape that the floats</param>
+        /// <param name="dims">The shape of values</param>
         /// <param name="name">A string used as a seed to generate this initializer's name in the ONNX graph.</param>
         /// <param name="makeUniqueName">Whether a unique name should be picked for this initializer.</param>
         /// <returns>The initializer's ONNX name</returns>
         public abstract string AddInitializer(IEnumerable<float> values, IEnumerable<long> dims, string name = null, bool makeUniqueName = true);
 
         /// <summary>
-        /// Call this function can declare a global long tensor
+        /// Call this function to declare a global long tensor
```

Comment:
This conflicts with the type param description: "Use this for adding array initializers of integer types smaller than Int32"